rakelib/license.rb

```diff
@@ -0,0 +1,40 @@
+# we need to call exit explicity  in order to set the proper exit code, otherwise
+# most common CI systems can not know whats up with this tests.
+require_relative "default_plugins"
+require "rubygems/specification"
+require "bootstrap/environment"
+
+
+def all_installed_gems
+  ENV["GEM_HOME"] = ENV["GEM_PATH"] = LogStash::Environment.logstash_gem_home
+  Gem.use_paths(LogStash::Environment.logstash_gem_home)
+
+  Gem::Specification.all = nil
```

Comment:
I don't understand the logic behind this set of method calls. Can you explain them to me? specially using the ```.all``` method recommends in the documentation:

```
Returns all specifications. This method is discouraged from use. You probably want to use one of the Enumerable methods instead.
```

I think is totally safe and more easy to use one of the many enumerated methods available. 

What do you think?